File: DSA-DAILY/31-07-2025/898. Bitwise ORs of Subarrays @Leetcode-Med.java
import java.util.*;
class Solution {
    public int subarrayBitwiseORs(int[] arr) {
        Set<Integer> res=new HashSet<>();
        Set<Integer> curr=new HashSet<>();
        for(int i:arr){
            Set<Integer> temp=new HashSet<>();
            temp.add(i);
            for(int j:curr){
                temp.add(i|j);
            }
            res.addAll(temp);
            curr=temp;
        }
        return res.size();
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter an array of numbers separated by spaces: ");
        String input = scanner.nextLine();
        int[] arr = Arrays.stream(input.split(" "))
                          .mapToInt(Integer::parseInt)
                          .toArray();
        Solution solution = new Solution();
        int result = solution.subarrayBitwiseORs(arr);
        System.out.println("Number of distinct bitwise ORs: " + result);
    }
}

File: DSA-DAILY/31-07-2025/898. Bitwise ORs of Subarrays @Leetcode-Med.py
from typing import List
class Solution:
    def subarrayBitwiseORs(self, arr: List[int]) -> int:
        res=set()
        for i in range(0,len(arr)):
            res.add(arr[i])
            for j in range(i-1,-1,-1):
                if(arr[j]==(arr[j]|arr[i])):
                    break
                arr[j]|=arr[i]
                res.add(arr[j])
        return len(res)        

if __name__ == "__main__":
    arr = list(map(int, input("Enter an array of numbers separated by spaces: ").split()))
    solution = Solution()
    result = solution.subarrayBitwiseORs(arr)
    print("Number of distinct bitwise ORs:", result)